### ISSUE TITLE
Consul server catalog health checks not working, agent API not sufficient

### ISSUE THREAD

**Issue by dakotasanchez** [hashicorp/consul#4709]
Hello,

From what I've gathered this is not exactly how consul is supposed to be used, but I'm still trying to make it work because it fits our use case (unless you can point me to a better configuration?).

Setup:
* 3 Consul server agents running (`consul agent -server -ui -blah -blah`) in a cluster.
* 3 consul-alerts nodes running, one beside each consul server agent.

Use case:
Have these consul nodes watching several http endpoints, and have consul-alerts notify us when a health check fails.

Problem:
If I register checks with the agent API (`consul-node-a:8500/v1/agent/service/register` or `...check/register`) then the health checks (and notifications) work just fine, but if `consul-node-a` goes down, then we lose the health checks with it. Not good, that's why we have a 3 node cluster, for high-availability. If I register the checks through each node to get around that, then I get duplicate notifications when each node sees the failed check. Also not good. If I register through the catalog API then it doesn't seem we have that availability problem, but then health checks don't work...

So basically, is there a way to make health checks work without using the agent API (since it can't handle node failure)? Should we be using consul ESM? Sort of at a deadend here. Thanks.

**Comment by mkeeler** [comment]
@dakotasanchez Issues on GitHub for Consul are intended to be related to bugs or feature requests, so we recommend using our other community resources for asking questions instead of asking here.

- [FAQ (Frequently asked questions)](https://www.consul.io/docs/faq.html)
- [Consul Guides](https://www.consul.io/docs/guides/index.html)
- Any other questions can be sent to the [consul mailing list](https://www.consul.io/community.html)

If you feel this is a bug, please open a [new issue](https://github.com/hashicorp/consul/issues/new?template=bug_report.md) with the appropriate information.

With that being said, it is working as expected. Usually you would run additional non server nodes and register the services/checks with them. Those health checks will still be tied to the health of those consul agents but just not the servers. Also because those agents are normally on the same node as the services being registered through them a failure of the node SHOULD indicate a failure in the other services on the same node.

If for some reason you can't run additional non-server agents then you should look into [consul-esm](https://github.com/hashicorp/consul-esm). It is for that exact scenario where you use the catalog to register services and checks but have consul-esm executing the health checks and posting the status back to the consul servers.
